docs/api/codegen.rst

```diff
@@ -0,0 +1,90 @@
+.. _edgedb-python-codegen:
+
+===============
+Code Generation
+===============
+
+.. py:currentmodule:: edgedb
+
+The ``edgedb-python`` package exposesa a command-line tool to generate
+typesafe functions using dataclasses from ``*.edgeql`` files.
+
+.. code-block:: bash
+
+  $ edgedb-py
+
+Consider a simple query that lives in a function called ``get_number.py``:
+
+.. code-block:: edgeql
+
+  select <int64>$arg;
+
+Running the code generator will generate a new file called
+``get_number_async_edgeql.py`` containing the following code (roughly):
+
+.. code-block:: python
+
+  from __future__ import annotations
+  import edgedb
+
+
+  async def get_number(
+      client: edgedb.AsyncIOClient,
+      *,
+      arg: int,
+  ) -> int:
+      return await client.query_single(
+          """\
+          select <int64>$arg\
+          """,
+          arg=arg,
+      )
+
+Target
+~~~~~~
+
+By default, the generated code uses an ``async`` API. The generator supports
+additional targets via the ``--target`` flag.
+
+.. code-block:: bash
+
+  $ edgedb-py --target async        # generate async function (default)
+  $ edgedb-py --target blocking     # generate blocking code
+  $ edgedb-py --target pydantic     # use pydantic typing
+
+The names of the generated files will differ accordingly:
+``{query_filename}_{target}_edgeql.py``.
```

Comment:
Oh in this case, the `pydantic` is a special target that doesn't really change the generated file name. I think I should probably use a CLI option like `--skip-pydantic-validation` rather than an option.